### DISCUSSION TITLE
How to get underscore accepted in dns hostname for txt record?

### DISCUSSION THREAD

**Discussion by john2893** [netbox-community/netbox#17155] [Help Wanted!] [answered]
I have a dns TXT record with name - _**xyz_test._ke.dom**_ is not a valid DNS host name.

I get the error xyz_test._ke.dom is not a valid DNS host name

Is there a setting I can enable? 

![ss1](https://github.com/user-attachments/assets/4370caf2-c7c6-4dcb-bb98-553310ec3526)

**Comment by candlerb** [comment, marked as answer]
That message doesn't appear anywhere in the Netbox source code, so I'm guessing you're using some unnamed third-party plugin. In that case, it would be best to take it up with the author of that plugin, or modify the source code yourself.

Aside: technically, what you have is a valid domain name, but not a valid hostname (according to an [ancient RFC](https://datatracker.ietf.org/doc/html/rfc952)).  However, there are plenty of DNS labels which are not used as hostnames, such as `_acme-challenge`, so I think your DNS plugin is in the wrong.

**Reply by john2893** [reply]
I had to add the below parameter in the plugin config

"tolerate_underscores_in_labels": True,
